### PR TITLE
[Mobile] Disable Page Template picker if Modal Layout Picker is enabled

### PR DESCRIPTION
**Pull request by antonis**
**Fixes:** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2419

**Related PRs:**
- `WordPress-Android`: https://github.com/wordpress-mobile/WordPress-Android/pull/12482
- `gutenberg-mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2494

## Description
Adds a capability to hide the editor layout picker when the Modal Layout Picker(MLP) is enabled.
For detailed description of this PR please check https://github.com/wordpress-mobile/WordPress-Android/pull/12482

## Notes
This has been updated with the [Consolidation of the Gutenberg initialProps handling](https://github.com/wordpress-mobile/WordPress-Android/pull/12444)

## How has this been tested?
This feature can be tested by using the build from https://github.com/wordpress-mobile/WordPress-Android/pull/12482

## Screenshots

|MLP Enabled|MLP Disabled|
|---|---|
|device-2020-07-22-180349|device-2020-07-22-175155|

## Types of changes
New feature

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/react-native-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/GutenbergProps.kt [hunk 1]
```diff
@@ -52,5 +54,6 @@ data class GutenbergProps(
         const val PROP_CAPABILITIES = "capabilities"
         const val PROP_CAPABILITIES_MENTIONS = "mentions"
         const val PROP_CAPABILITIES_UNSUPPORTED_BLOCK_EDITOR = "unsupportedBlockEditor"
+        const val PROP_NAME_MODAL_LAYOUT_PICKER = "modalLayoutPicker"
```
Comment (mchowning) [inline packages/react-native-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/GutenbergProps.kt:57]: Not a big deal, but for the sake of consistency with the other "capabilities" props, what would you think about renaming this variable to "PROP_CAPABILITIES_MODAL_LAYOUT_PICKER"?

Comment (antonis) [reply packages/react-native-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/GutenbergProps.kt]: You are right @mchowning . I copy pasted the wrong naming from the old code.
Thank you for your feedback :)

**Review by chipsnyder — APPROVED** [review]
LGTM! Thanks for making that naming change.

File: packages/react-native-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/GutenbergProps.kt [hunk 2]
```diff
@@ -10,7 +10,8 @@ data class GutenbergProps(
     val editorTheme: Bundle?,
     val translations: Bundle,
     val isDarkMode: Boolean,
-    val htmlModeEnabled: Boolean
+    val htmlModeEnabled: Boolean,
+    val isModalLayoutPickerEnabled: Boolean
```
Comment (antonis) [inline packages/react-native-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/GutenbergProps.kt:14]: I was thinking of adding a default value to the new parameter in order to be able to merge this early without breaking the WP-Android code. Something like the following:

```
data class GutenbergProps @JvmOverloads constructor(
    val enableMentions: Boolean,
    val enableUnsupportedBlockEditor: Boolean,
    val localeSlug: String,
    val postType: String,
    val editorTheme: Bundle?,
    val translations: Bundle,
    val isDarkMode: Boolean,
    val htmlModeEnabled: Boolean
    val htmlModeEnabled: Boolean,
    val isModalLayoutPickerEnabled: Boolean = false
) {
```

Wdyt @mchowning , @chipsnyder ?

Comment (chipsnyder) [reply packages/react-native-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/GutenbergProps.kt:14]: That seems like a good plan to me.

**Pull request merged**